src/shared/userdb.c

```diff
@@ -1347,16 +1341,13 @@ int groupdb_by_name(const char *name, const UserDBMatch *match, UserDBFlags flag
         }
 
         /* As above, we apply our own client-side filtering even if server-side filtering worked, for robustness and simplicity reasons. */
-        r = group_record_match(gr, match);
-        if (r < 0)
-                return r;
-        if (r == 0)
+        if (!group_record_match(gr, match))
                 return -ENOEXEC;
 
         if (ret)
                 *ret = TAKE_PTR(gr);
 
-        return r;
+        return 1;
```

Comment:
Should be able to return 0 here

Reply:
I concur (all calllers I could see don't distinguish these two). Added as a separate fix.